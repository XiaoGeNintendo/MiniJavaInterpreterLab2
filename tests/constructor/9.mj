class Fa{
    int t=3;
    int q=(this.t+=3);
    int fa=-2;
}
class X extends Fa{
    X(){
        this(1);
        println(t+" "+q);
    }

    X(int x){
        println(t+" "+q);
    }
}

class Fa2{
    int q=(this.t+=3);
    int t=3;
    int fa=-2;
}
class X2 extends Fa2{
    X2(){
        this(1);
        println(t+" "+q);
    }

    X2(int x){
        println(t+" "+q);
    }
}

int main(){
    new X();
    new X2();
    return 0;
}