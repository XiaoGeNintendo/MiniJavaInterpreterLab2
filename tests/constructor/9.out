6 6
6 6
3 3
3 3
Process exits with the code 0.